### PR TITLE
Feature hls streaming

### PR DESCRIPTION
**Pull request by Arunshaik2001**

### CODE REVIEW

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt [hunk 1]
```diff
@@ -769,9 +811,9 @@ class HmssdkFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
             previewChannel.setStreamHandler(null)
             val args = HashMap<String, Any?>()
             args.put("event_name", "on_join_room")
-
+            Log.i("onjoinAndroidPreviewListener", room.hlsStreamingState?.running.toString())
```
Comment (jashan1498) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt:814]: @Arunshaik2001 remove

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt [hunk 2]
```diff
@@ -907,12 +952,11 @@ class HmssdkFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
             val args = HashMap<String, Any?>()
             args.put("event_name", "preview_video")
             args.put("data", HMSPreviewExtension.toDictionary(room, localTracks))
-//        Log.i("onPreview", args.get("data").toString())
+            Log.i("onPreviewAndroidPreviewListener", room.hlsStreamingState.toString()+room.peerList.size)
```
Comment (jashan1498) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt:955]: remove @Arunshaik2001

File: example/lib/meeting/meeting_store.dart [hunk 3]
```diff
@@ -103,14 +98,14 @@ abstract class MeetingStoreBase extends ChangeNotifier
   @action
   void startListen() {
     meetingController.addMeetingListener(this);
-    // startHMSLogger(HMSLogLevel.VERBOSE, HMSLogLevel.VERBOSE);
-    // addLogsListener();
+    //startHMSLogger(HMSLogLevel.DEBUG, HMSLogLevel.DEBUG);
```
Comment (jashan1498) [inline example/lib/meeting/meeting_store.dart:101]: remove if not being used here anymore

File: example/lib/meeting/meeting_store.dart [hunk 4]
```diff
@@ -256,18 +251,27 @@ abstract class MeetingStoreBase extends ChangeNotifier
   @override
   void onJoin({required HMSRoom room}) async {
     hmsRoom = room;
+    if(room.hmshlsStreamingState?.running??false){
+      hasHlsStarted=true;
+      streamUrl = room.hmshlsStreamingState?.variants[0]?.hlsStreamUrl??"";
```
Comment (jashan1498) [inline example/lib/meeting/meeting_store.dart:256]: what happens if streamUrl is Empty ?

File: example/lib/meeting/meeting_store.dart [hunk 5]
```diff
@@ -585,7 +566,7 @@ abstract class MeetingStoreBase extends ChangeNotifier
   void leaveMeeting() async {
     meetingController.leaveMeeting();
     isRoomEnded = true;
-    // removeListenerMeeting();
+    //removeListenerMeeting();
```
Comment (jashan1498) [inline example/lib/meeting/meeting_store.dart:569]: remove

File: example/lib/meeting/meeting_store.dart (hunk 4 above)
Comment (Arunshaik2001) [reply example/lib/meeting/meeting_store.dart:256]: If running is true then why stream url would be empty?

**Pull request closed without merging**